express-notes-api/server.js

```diff
@@ -0,0 +1,118 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  const array = [];
```

Comment:
You must put try/catch around each of these handler functions. If `readJSONData` throws an exception you won't catch it and the client will not get a proper error response.